public/app/features/search/service/sql.ts

```diff
@@ -70,6 +71,40 @@ export class SQLSearcher implements GrafanaSearcher {
     return this.doAPIQuery(q);
   }
 
+  async starred(query: SearchQuery): Promise<QueryResponse> {
+    if (query.facet?.length) {
+      throw 'facets not supported!';
+    }
+    const q: APIQuery = {
+      limit: query.limit ?? 1000, // default 1k max values
+      tag: query.tags,
+      sort: query.sort,
+      starred: query.starred,
+    };
+
+    query = await replaceCurrentFolderQuery(query);
+    if (query.query === '*') {
+      if (query.kind?.length === 1 && query.kind[0] === 'folder') {
```

Comment:
trivial. should we combine this with the first `if` condition?

Reply:
If we combine the first condition, we're changing the logic, don't we?. For example:
```
// This is how it is
if(A) {
   if(B){ 
    foo() 
  }
} else if (C){
bar()
}
```
bar() will execute if !A && C

Instead, if we change to 

```
// This is how it is
if(A && B) {
    foo() 
} else if (C){
  bar()
}
```
bar() will be executed if !(A && B) which is different logic than before, because B value doesn't affect the equation. But if we apply your suggestion, B must be true to execute `bar()`.

Did I explain myself? 😓